src/redux/books/bookSlice.js

```diff
@@ -0,0 +1,15 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const bookSlice = createSlice({
+  name: 'books',
+  initialState: [],
+  reducers: {
+    bookAdded(state, action) {
+      state.push(action.payload);
+    },
+  },
+});
+
+export const selectBooks = (state) => state.books;
+export const { bookAdded } = bookSlice.actions;
+export default bookSlice.reducer;

```

Comment:
Your `books` reducer is missing `action types` and `action creators` and the only thing that should be exported is the reducer.
- Action types are simply constants/variables that redux uses to reference a particular action.
- An action creator is a function that literally creates an action object. In Redux, action creators simply return an action object and pass the argument value if necessary. The object often contains the data/response data that will be saved as the new state.

Please follow this structure to set up your books reducer property according to the project requirements:
![image](https://user-images.githubusercontent.com/38088818/167385089-a26a2bea-60f1-4571-8fdb-0ed60a0e1527.png)

Create action types:
```
const ADD_BOOK = bookstore/books/ADD_BOOK
const REMOVE_BOOK = bookstore/books/REMOVE_BOOK
```

Add action creators:
```
export const joinMission = (payload) => ({
  type: ADD_BOOK,
  payload,
});

export const leaveMission = (payload) => ({
  type: REMOVE_BOOK,
  payload,
});
```

Next, kindly set up your reducer properly for both actions:
```
const booksReducer = (state = [], action) => {
  switch (action.type) {
    case ADD_BOOK:
      return [...state, action.payload];

    case REMOVE_BOOK:
      return state.filter((book) => book.item_id !== action.payload);

    default:
      return state;
  }
};
```

Then, export reducer as default:
```
export default booksReducer;
```